### ISSUE TITLE
Type of `ObservableStatus::data` should include `undefined`

### ISSUE THREAD

**Issue by wmadden** [FirebaseExtended/reactfire#577]
### Version info

**React:**

**Firebase:**

**ReactFire:**
4.2.2

**Other (e.g. Node, browser, operating system) (if applicable):**

### Test case

N/A

**Comment by devth** [comment]
Agree. I've always found the inaccurate types confusing. It leads to error prone code, since the result of these hooks is briefly `undefined` before they return valid values.

**Comment by webbertakken** [comment]
It's on their radar now: https://github.com/FirebaseExtended/reactfire/issues/570#issuecomment-4772709097
